task-1/app.py

```diff
@@ -0,0 +1,97 @@
+from flask import Flask, jsonify, request
+import json
+import http
+
+class InterfaceAPI(Flask):
+
+    FILE_NAME = "data.json"
+    FILTERED_OPTIONS = ["name", "type", "enabled"]
+
+    def __init__(self, import_name):
+        super(InterfaceAPI, self).__init__(import_name)
+        self.cache = self.read_json(self.FILE_NAME) 
+
+    def read_json(self, file_name):
+        with open(file_name) as f:
+            data = json.loads(f.read())
+            interfaces = data['ietf-interfaces:interfaces']['interface']
+            return {interface['name']: interface for interface in interfaces}
+
+app = InterfaceAPI(__name__)
+
+@app.route('/get-all-interfaces/', methods=["GET"])
+def view_all_interfaces():
+    interfaces = app.cache.values()
+    if interfaces:
+        return jsonify(list(interfaces)), http.HTTPStatus.OK
+    return http.HTTPStatus.NOT_FOUND
+
+@app.route('/get-all-interfaces/', methods=["HEAD"])
+def view_all_interfaces_head():
+    interfaces = app.cache.values()
+    if interfaces:
+        return http.HTTPStatus.OK
+    return http.HTTPStatus.NOT_FOUND
+
+@app.route('/get-interface/<path:interface_name>', methods=["GET"])
+def view_interface_by_name(interface_name):
+    interface = app.cache.get(interface_name)
+    if interface:
+        return jsonify(interface), http.HTTPStatus.OK
+    return jsonify({"info": "interface not found"}), http.HTTPStatus.NOT_FOUND
+    
+@app.route('/get-interface/<path:interface_name>', methods=["HEAD"])
+def view_interface_by_name_head(interface_name):
+    interface = app.cache.get(interface_name)
+    if interface:
+        return http.HTTPStatus.OK
+    return http.HTTPStatus.NOT_FOUND
+
+def validation(option, value):
+    for item in option:
+        key = list(item.keys())[0]
```

Comment:
@tomowex option is single data.. like name: something,
type something.. and then you do for item in option which is single key.. or do I understand this incorectly? what do you expect here as an option and value?